pkg/api/v1alpha1/cluster_types_test.go

```diff
@@ -3230,3 +3230,37 @@ func TestValidateCluster(t *testing.T) {
 		})
 	}
 }
+
+func TestCluster_SetManagmentComponentsVersion(t *testing.T) {
+	managmentClusterName := "mgmt-cluster"
+	expectedManagmentComponentVersion := "v0.0.0-dev+build.0000"
+
+	testCases := []struct {
+		name            string
+		cluster         *v1alpha1.Cluster
+		wantAnnotations map[string]string
+	}{
+		{
+			name: "self-managed cluster",
+			cluster: baseCluster(func(c *v1alpha1.Cluster) {
+				c.SetManagmentComponentsVersion(expectedManagmentComponentVersion)
+			}),
+			wantAnnotations: map[string]string{"anywhere.eks.amazonaws.com/eksa-management-components-version": expectedManagmentComponentVersion},
+		},
+		{
+			name: "managed cluster",
+			cluster: baseCluster(func(c *v1alpha1.Cluster) {
+				c.SetManagedBy(managmentClusterName)
+				c.SetManagmentComponentsVersion(expectedManagmentComponentVersion)
```

Comment:
I think the test would be more clear if you called this in the body of the test instead of the test case
it makes easier to understand what you are testing
The test case specifies input and out output
The test body passed the input through a method and compares the real output with the expected output.

Reply:
Updated